opentelemetry/src/sdk/trace/span.rs

```diff
@@ -78,9 +79,18 @@ impl crate::trace::Span for Span {
         &mut self,
         name: String,
         timestamp: SystemTime,
-        attributes: Vec<KeyValue>,
+        mut attributes: Vec<KeyValue>,
     ) {
+        let max_attributes_per_event =
+            self.tracer
+                .provider()
+                .map(|provider| provider.config().max_attributes_per_event)
+                .unwrap_or(DEFAULT_MAX_ATTRIBUTES_PER_EVENT) as usize;
         self.with_data(|data| {
+            if attributes.len() > max_attributes_per_event {
+                let _dropped: Vec<_> = attributes.drain((max_attributes_per_event)..).collect();
+            }
```

Comment:
I think you should be able to replace these 3 lines with a call to [`truncate`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.truncate):

```suggestion
            attributes.truncate(max_attributes_per_event);
```